### ISSUE TITLE
Monolithic SST Stats API

### ISSUE THREAD

**Issue by jleidel** [sstsimulator/sst-core#243]
I've been working on an enhancement for the HMC-Sim library implementation that directly integrates the SST Statistics API into the HMC-Sim core.  I was previously under the assumption that the SST Statistics could be created/registered/utilized by external libraries.  However, in looking through the actual source, there doesn't appear to be an separate SST Stats library to link against, which implies that the SST Stats are monolithically integrated into the SST Core.  Put simply, do I have the ability to directly integrate Stats into the HMC-Sim (libhmcsim.so) library?  (as opposed to the GC64-HMC-Sim memBackend)

**Comment by jleidel** [comment]
Related to this, I've been using the tutorial at the link below as the basis for my modifications.  I've attempted to integrate some basic statistics directly in the goblinHMCBackend memBackend.  The tutorial points at the simpleElementExample source as the basis for comparison.  However, the simpleElementExample appears to make heavy use of macros when initializing the Stats API as opposed to the syntax/methods found in the tutorial.  Which is correct?  My current code follows the source from the tutorial and my example fails to compile with type issues: 

membackend/goblinHMCBackend.cc:22:14: error: ‘ElementInfoStatistic’ does not name a type
 static const ElementInfoStatistic GoblinHmcsim_statistics[] = {

I've tried including a number of different headers from SST-Core, but none appear to fix the issue.    

http://sst-simulator.org/SSTPages/SSTDeveloperSSTStatisticsAPI/
